sormas-backend/src/main/java/de/symeda/sormas/backend/contact/ContactFacadeEjb.java

```diff
@@ -1362,11 +1362,26 @@ private void restorePseudonymizedDto(ContactDto dto, Contact existingContact, Co
 		if (existingContactDto != null) {
 			boolean isInJurisdiction = contactJurisdictionChecker.isInJurisdictionOrOwned(existingContact);
 			User currentUser = userService.getCurrentUser();
+
+			String followUpComment = null;
+			if (dto.isPseudonymized()) {
```

Comment:
I don't see this to be a good solution, the followup comment can be changed through the REST API. It could work as a hotfix but we need a new change request ticket for properly handling system generated comments. Maybe store them in a separate column in the db.

the condition `dto.isPseudonymized()` is always not enough, you should check also if the followup field is accessible or not; see the condition in DtoPseudonymizer::restorePseudonymizedValues 
```java
if (!getFieldAccessCheckers(isInJurisdiction).isAccessible(pseudonymizedField, pseudonymizeMandatoryFields) || dto.isPseudonymized()) {
   restoreOriginalValue(dto, pseudonymizedField, originalDto);
}
````

The accessibility check is needed when someone edits a contact that gets out of jurisdiction in the mean time. In this case it will not work to save the followup comment from the DTO and add it after restore because it will duplicate the comment text.

Similar thing should be done for cases too.

Reply:
I added a field accessibility check.

I could not find system follow up comments for cases, that's why there is nothing similar for cases.

I am also not very happy with this solution. Why are changes via the REST API a problem? They call the same method...

Reply:
Changes via the REST API could have been a problem without checking the field access. There could have been a similar issue I mentioned in my previous comment: 

> The accessibility check is needed when someone edits a contact that gets out of jurisdiction in the mean time. In this case it will not work to save the followup comment from the DTO and add it after restore because it will duplicate the comment text.